tests/system/src/test/java/test/javafx/stage/FocusedWindowTest.java

```diff
@@ -0,0 +1,142 @@
+    static Stage stage = null;
+
+    static {
+        System.setProperty("glass.platform","Monocle");
+        System.setProperty("monocle.platform","Headless");
+    }
+
+    public static class TestApp extends Application {
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            primaryStage.setTitle("Primary Stage");
+            primaryStage.setScene(new Scene(new TextField()));
+
+            primaryStage.setOnShown(l -> {
+                Platform.runLater(() -> startupLatch.countDown());
+            });
+            primaryStage.show();
+            Platform.setImplicitExit(false);
+            stage = primaryStage;
+        }
+    }
+
+    @BeforeClass
+    public static void initFX() throws Exception {
+        startupLatch = new CountDownLatch(1);
+        new Thread(() -> Application.launch(TestApp.class, (String[]) null)).start();
+        Assert.assertTrue("Timeout waiting for FX runtime to start", startupLatch.await(15, TimeUnit.SECONDS));
+        Platform.runLater(() -> stage.close());
+    }
+
+    @Test
+    public void testLeak() throws Exception {
+        int counter = 0;
+        while(counter <= 100) {
+            counter += 1;
+            testLeakOnce();
+        }
+    }
+
+    static WeakReference<Stage> closedFocusedStageWeak = null;
+
+    public void testLeakOnce() throws Exception {
+        CountDownLatch leakLatch = new CountDownLatch(1);
+        closedFocusedStageWeak = null;
+        Platform.runLater(() -> {
+            Stage closedFocusedStage = new Stage();
+            closedFocusedStage.setTitle("Focused Stage");
+            closedFocusedStageWeak = new WeakReference<>(closedFocusedStage);
+            TextField textField = new TextField();
+            closedFocusedStage.setScene(new Scene(textField));
+            Platform.runLater(() -> {
+                closedFocusedStage.show();
+                Platform.runLater(() -> {
+                    closedFocusedStage.close();
+                    Platform.runLater(() -> {
+                        closedFocusedStage.requestFocus();
+                        //textField.requestFocus();
+                        Platform.runLater(() -> {
+                            leakLatch.countDown();
+                        });
+                    });
+                });
+            });
+        });
+
+        Assert.assertTrue("Timeout, waiting for runLater. ", leakLatch.await(15, TimeUnit.SECONDS));
+
+        assertCollectable(closedFocusedStageWeak);
+
+    }
+
+    public static void assertCollectable(WeakReference weakReference) throws Exception {
+        int counter = 0;
+
+        System.gc();
+        System.runFinalization();
+
+        while (counter < 10 && weakReference.get() != null) {
+            Thread.sleep(100);
+            counter = counter + 1;
+            System.gc();
+            System.runFinalization();
+        }
+
+        Assert.assertNull(weakReference.get());
```

Comment:
Unifying memory-tests is something I discussed in my last PR. This code is based on my previous pull request: https://github.com/openjdk/jfx/pull/71
I would like to keep this version which I've used there. But it really has to be a library/utility-class which is used across the project instead of reinventing it on every test.